Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m

```diff
@@ -75,19 +75,31 @@ - (BUYRequestOperation *)getProductsPage:(NSUInteger)page completion:(BUYDataPro
 	}];
 }
 
-- (BUYRequestOperation *)getProductById:(NSString *)productId completion:(BUYDataProductBlock)block;
+- (BUYRequestOperation *)getProductByHandle:(NSString *)handle completion:(BUYDataProductBlock)block
+{
+	BUYAssert(handle, @"Failed to get product by handle. Product handle must not be nil.");
+	
+	NSURL *url = [self urlForProductListingsWithParameters:@{ @"handle" : handle }];
+	return [self getRequestForURL:url completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
+		
+		BUYProduct *product = nil;
+		if (json && !error) {
+			product = [self.modelManager insertProductsWithJSONArray:json[BUYProductsKey]].firstObject;
+		}
+		
+		if (!product && !error) {
```

Comment:
If the `handle` provided isn't found, the server will return an empty collection. In this case, that means the `product` and `error` will be `nil`.